From: Nicolas Saenz Julienne <nsaenzjulienne@suse.de>
Date: Wed, 21 Oct 2020 18:34:21 +0200
Subject: gpio: raspberrypi-exp: Release firmware handle on unbind
References: jsc#SLE-16543
Git-commit: 0e3333b29cad8b8906a7bc6af61f7e0df8cfd3b5
Patch-mainline: v5.13-rc1

Use devm_rpi_firmware_get() so as to make sure we release RPi's firmware
interface when unbinding the device.

Signed-off-by: Nicolas Saenz Julienne <nsaenzjulienne@suse.de>
Acked-by: Bartosz Golaszewski <bgolaszewski@baylibre.com>

---

Changes since v2:
 - Use devm_rpi_firmware_get(), instead of remove function
---
 drivers/gpio/gpio-raspberrypi-exp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpio/gpio-raspberrypi-exp.c b/drivers/gpio/gpio-raspberrypi-exp.c
index bb100e0124e6..64a552ecc2ad 100644
--- a/drivers/gpio/gpio-raspberrypi-exp.c
+++ b/drivers/gpio/gpio-raspberrypi-exp.c
@@ -208,7 +208,7 @@ static int rpi_exp_gpio_probe(struct platform_device *pdev)
 		return -ENOENT;
 	}
 
-	fw = rpi_firmware_get(fw_node);
+	fw = devm_rpi_firmware_get(&pdev->dev, fw_node);
 	of_node_put(fw_node);
 	if (!fw)
 		return -EPROBE_DEFER;

